networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -415,6 +476,30 @@ def get_metagroups(self, connected_devices: Set[NbRecord]) -> List[conf.Hostgrou
             metagroup.members.sort()  # type: ignore
         return list(metagroups.values())
 
+    @classmethod
+    def purge_infranetworks_on_metagroup(cls, metagroup: conf.Hostgroup, member_hgs: List[conf.Hostgroup]):
+        if not metagroup.metagroup:
+            raise ValueError('Argument metagroup must be a metagroup')
+        metagroup_members = set(metagroup.members)
+        members_hgs_binding_hosts = set(x.binding_hosts[0] for x in member_hgs)
+        if metagroup_members != members_hgs_binding_hosts:
+            raise ValueError("'member_hgs.binding_host[0]' and metagroup.members are not matching")
+
+        minimum_common_infra_nets = set(member_hgs[0].infra_networks)
+        for member in member_hgs:
+            minimum_common_infra_nets.intersection_update(member.infra_networks)
+
+        # purge them from the member_hgs
+        for member in member_hgs:
+            remaining_infra_nets = list()
+            for infra_net in member.infra_networks:
+                if infra_net not in minimum_common_infra_nets:
+                    remaining_infra_nets.append(infra_net)
+            member.infra_networks = remaining_infra_nets
+        metagroup.infra_networks = sorted(minimum_common_infra_nets, key=lambda x: x.vlan)
+
+        return metagroup, member_hgs
```

Comment:
I usually try to explicitly return even if the ops was in-place. I will consume the return values from the function, deal?

Reply:
Hmm to be honest I think I like the way you have it, no need to consume the return values if it would be the same without them. I like references and in-place ops.